database/database_test.go

```diff
@@ -234,3 +234,180 @@ func TestStatesWithStatus(t *testing.T) {
 		}
 	}
 }
+
+// setTests is a struct that is used in testing SetIneligibleDomainStates
+var setTests = []struct {
+	description       string
+	domainStates      []IneligibleDomainState
+	wantStatusReports []string
+}{
+	{
+		"two domains",
+		[]IneligibleDomainState{
+			{Name: "youtube.com", Policy: "bulk-1-year"},
+			{Name: "garron.net", Policy: "bulk-1-year"},
+		},
+		[]string{"Updating 2 entries...", " done.\n"},
+	},
+	{
+		"bulk-18-week",
+		[]IneligibleDomainState{{Name: "gmail.com", Policy: "bulk-18-week"}},
+		[]string{"Updating 1 entries...", " done.\n"},
+	},
+	{
+		"bulk-1-year",
+		[]IneligibleDomainState{{Name: "wikipedia.com", Policy: "bulk-1-year"}},
+		[]string{"Updating 1 entries...", " done.\n"},
+	},
+}
+
+// Test SetIneligibleDomainStates
+func TestSetIneligibleDomainStates(t *testing.T) {
+	resetDB()
+
+	for _, tt := range setTests {
+
+		var statuses []string
+		statusReport := func(format string, args ...interface{}) {
+			formatted := fmt.Sprintf(format, args...)
+			statuses = append(statuses, formatted)
+		}
+
+		err := testDB.SetIneligibleDomainStates(
+			tt.domainStates,
+			statusReport,
+		)
+		if err != nil {
+			t.Errorf("[%s] cannot put states %s", tt.description, err)
+			return
+		}
+
+		if !reflect.DeepEqual(statuses, tt.wantStatusReports) {
+			t.Errorf("[%s] Incorrect status reports: %#v", tt.description, statuses)
+		}
+	}
+}
+
+// getAndDeleteTests is a struct that is used in testing GetIneligibleDomainStates
+// and DeleteIneligibleDomainStates
+var getAndDeleteTests = []struct {
+	description  string
+	domainStates []string
+	wantStates   []IneligibleDomainState
+}{
+	{
+		"one domain",
+		[]string{"a.com"},
+		[]IneligibleDomainState{
+			{Name: "a.com", Policy: "bulk-1-year"},
```

Comment:
Total drive-by-nit, but I would recommend against using any test domains names that are registerable, even if the real domains are not meant accessed in tests. (In fact, the HSTS preload list itself has caused issues for people who assumed it was safe to use test domains that could and did actually become registerable.)

`.test` and `.example` are safe choices: https://www.rfc-editor.org/rfc/rfc2606.html#section-2

Reply:
This is very interesting and something I hadn't thought about. I went ahead and fixed it! Thank you for bringing this up